examples/chip-tool/templates/tests/partials/saveAs/maybeSaveAs.zapt

```diff
@@ -1,12 +1,16 @@
 {{#if saveAs}}
     {{#if (isString type)}}
-        if ({{saveAs}}Buffer != nullptr)
-        {
-            chip::Platform::MemoryFree({{saveAs}}Buffer);
-        }
-        {{saveAs}}Buffer = static_cast<{{#if (isOctetString type)}}uint8_t{{else}}char{{/if}} *>(chip::Platform::MemoryAlloc({{asPropertyValue}}.size()));
-        memcpy({{saveAs}}Buffer, {{asPropertyValue}}.data(), {{asPropertyValue}}.size());
-        {{saveAs}} = {{chipType}}({{saveAs}}Buffer, {{asPropertyValue}}.size());
+        {{#if isNullable}}
+            {{saveAs}} = {{asPropertyValue}};
```

Comment:
No, this is not ok.  This will just store the pointer, but the pointer is pointing into memory that's about to be deallocated.

What needs to happen here for the nullable case is:

1. If null, just set saved value to null.
2. If not null, do the copying thing, but using `{{asPropertyValue}}.Value()` as the span.  Should be doable with an inline partial for the span that conditions on isNullable.  Then set `{{saveAs}}` to a value pointing at our buffer.

Reply:
Ok. Thank you for the help on this one. Updated.